File: Backend/src/main/java/com/luckyseven/backend/domain/settlements/dto/SettlementResponse.java
package com.luckyseven.backend.domain.settlements.dto;

import com.luckyseven.backend.domain.settlements.entity.Settlement;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import lombok.Builder;

/**
 * DTO for {@link Settlement}
 */
@Builder
public record SettlementResponse(
    Long id,
    LocalDateTime createdAt,
    LocalDateTime updatedAt,
    BigDecimal amount,
    Boolean isSettled,
    Long settlerId,
    String settlerNickname,
    Long payerId,
    String payerNickname,
    Long expenseId,
    String expenseDescription,
    Long teamId
) {

}

File: Backend/src/main/java/com/luckyseven/backend/domain/settlements/dto/SettlementResponse.kt
package com.luckyseven.backend.domain.settlements.dto

import java.math.BigDecimal
import java.time.LocalDateTime

data class SettlementResponse(
    val id: Long,
    val amount: BigDecimal,
    val isSettled: Boolean,
    val settlerId: Long,
    val settlerName: String,
    val payerId: Long,
    val payerName: String,
    val expenseId: Long,
    val expenseName: String,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
)